Sources/Sentry/SentryNetworkTracker.m

```diff
@@ -265,6 +288,114 @@ - (void)urlSessionTask:(NSURLSessionTask *)sessionTask setState:(NSURLSessionTas
+
+    SentryThreadInspector *threadInspector = SentrySDK.currentHub.getClient.threadInspector;
+    NSArray<SentryThread *> *threads = [threadInspector getCurrentThreads];
+
+    SentryException *sentryException = [[SentryException alloc] initWithValue:message
+                                                                         type:@"HTTP-ClientError"];
+    sentryException.mechanism =
+        [[SentryMechanism alloc] initWithType:@"SentryNetworkTrackingIntegration"];
+
+    if (threads.count > 0) {
+        for (SentryThread *thread in threads) {
+            if (nil != thread.current && [thread.current boolValue]) {
+                SentryStacktrace *sentryStacktrace = [thread stacktrace];
+                sentryStacktrace.snapshot = @(YES);
+
+                sentryException.stacktrace = sentryStacktrace;
+
+                break;
+            }
+        }
+    }
+
+    SentryRequest *request = [[SentryRequest alloc] init];
+
+    NSURLRequest *myRequest = (NSURLRequest *)sessionTask.currentRequest;
+
+    NSURL *url = [[sessionTask currentRequest] URL];
+
+    NSString *urlString = url.absoluteString;
+
+    if (nil != url.fragment) {
+        request.fragment = url.fragment;
+        urlString = [urlString
+            stringByReplacingOccurrencesOfString:[@"#" stringByAppendingString:url.fragment]
+                                      withString:@""];
+    }
+
+    if (nil != url.query) {
+        request.queryString = url.query;
+        urlString =
+            [urlString stringByReplacingOccurrencesOfString:[@"?" stringByAppendingString:url.query]
+                                                 withString:@""];
+    }
+
+    request.url = urlString;
+    request.method = myRequest.HTTPMethod;
+    if (sessionTask.countOfBytesSent != 0) {
+        request.bodySize = [NSNumber numberWithLongLong:sessionTask.countOfBytesSent];
+    }
+    if (nil != myRequest.allHTTPHeaderFields) {
+        NSDictionary<NSString *, NSString *> *headers = myRequest.allHTTPHeaderFields.copy;
+        request.headers = headers;
+        request.cookies = headers[@"Cookie"];
+    }
+
+    event.exceptions = @[ sentryException ];
+    event.request = request;
+
+    NSMutableDictionary<NSString *, id> *context = [[NSMutableDictionary alloc] init];
+    ;
+    NSMutableDictionary<NSString *, id> *response = [[NSMutableDictionary alloc] init];
+
+    [response setValue:responseStatusCode forKey:@"status_code"];
+    if (nil != myResponse.allHeaderFields) {
+        NSDictionary<NSString *, NSString *> *headers = myResponse.allHeaderFields.copy;
+        [response setValue:headers forKey:@"headers"];
+        [response setValue:headers[@"Cookie"] forKey:@"cookies"];
+    }
+    if (sessionTask.countOfBytesReceived != 0) {
+        [response setValue:[NSNumber numberWithLongLong:sessionTask.countOfBytesReceived]
+                    forKey:@"body_size"];
+    }
+
+    context[@"response"] = response;
```

Comment:
I noticed that Contexts is just a Dict of String, Any, but other SDKs have a statically typed data class for known contexts such as `app`, `os`, etc, why don't we have them in Cocoa?

Reply:
Not sure, this is legacy decision.

Reply:
Ideas @philipphofmann ? should we raise an issue for that?

Reply:
No real reason. We could change this in the next major. https://github.com/getsentry/sentry-cocoa/issues/2319